### PR TITLE
Add Triton + FIL backend, GKE horizontal autoscale example

### PR DESCRIPTION
**Pull request by drobison00**

### CODE REVIEW

**Comment by drobison00** [comment]
Tested the full end to end workflow. Everything works for HTTP; gRPC client throws an error that I haven't tracked down yet; maybe a bug with gRPC, or a configuration problem. Other than that, everything else is finished.

**Comment by zronaghi** [comment]
@mengdong can you please review?

File: triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml
Comment (mengdong) [inline triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml:41]: Did we test XGboost inference can reach 85% of GPU utilization and meet the typical SLA requirement? My concern is that latency could be large before HPA autoscale the Triton deployment

Comment (mengdong) [reply triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml:41]: and I caught my own mistake here, instead of 85, it should be:  `{{ .Values.HPATargetAverageValue }} `

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (mengdong) [inline triton/kubernetes/FIL/interact_with_triton.ipynb:48]: Just wanted to note that with my Helm chart, due to the way I set up Istio gateway and virtual service, the Triton deployment can only accept HTTP or GRPC format depends on the config of helm chart.

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (mengdong) [inline triton/kubernetes/FIL/interact_with_triton.ipynb:237]: It might be interesting to add a bit of benchmark result so the user can understand the benefits of GPU when serving XGBoost through Triton?

**Review by mengdong — COMMENTED** [review]
Awesome work to enable FIL backend of Triton in GKE. I've added a few minor suggestions and caught one of my own mistakes. 

Also suggest we add FIL backend to NGC public release of Triton so that users don't have to build their container everytime.

Thanks!

File: triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml
Comment (drobison00) [reply triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml:41]: I haven't verified that auto-scaling triggers yet, my account doesn't have the permissions to deploy the stackdriver metrics. Hopefully soon.

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (drobison00) [reply triton/kubernetes/FIL/interact_with_triton.ipynb:48]: Yes, thanks for the reminder. Was planning to update this so either could be used.

Comment (drobison00) [reply triton/kubernetes/FIL/interact_with_triton.ipynb:62]: Updated to support both in the same deployment.

Comment (mengdong) [reply triton/kubernetes/FIL/interact_with_triton.ipynb:62]: Thank you, Devin! I see you used the default TCP port for GRPC, could you please let me if TCP port/protocol is working in terms of loading balance GRPC traffic? For example, if you have 1 GRPC client and send all the requests.

File: triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml
Comment (zronaghi) [reply triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml]: Waiting on ops approval https://github.com/rapidsai/ops/issues/1554

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (zronaghi) [reply triton/kubernetes/FIL/interact_with_triton.ipynb:251]: Great point Dong, will add additional content after verifying auto-scaling

File: triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml
Comment (zronaghi) [reply triton/kubernetes/FIL/helm/chart/triton/templates/hpa.yaml]: @drobison00 spoke with Ray, we'll hold off on this for now. Anything else needed for this PR? If not, I'll merge and we can add benchmarks + auto-scaling later

File: dask/kubernetes/Dockerfile
Comment (Nanthini10) [inline dask/kubernetes/Dockerfile:1]: Update to latest version

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (Nanthini10) [inline triton/kubernetes/FIL/interact_with_triton.ipynb:260]: Just making sure - is this assertion error expected?

**Review by Nanthini10 — CHANGES_REQUESTED** [review]
Sorry for the late review. Looks good to me - just a version update!

File: dask/kubernetes/Dockerfile
Comment (zronaghi) [reply dask/kubernetes/Dockerfile:1]: Can we just actually update to 21.06 then?

File: triton/kubernetes/FIL/interact_with_triton.ipynb
Comment (drobison00) [reply triton/kubernetes/FIL/interact_with_triton.ipynb]: It was at the time, because it wasn't able to compare grpc and tcp approaches correctly. Should be fixed now.

**Review by zronaghi — APPROVED** [review]

**Comment by zronaghi** [comment]
@Nanthini10 can you give this another review, pending approval since your requested changes

**Review by Nanthini10 — APPROVED** [review]
Lgtm!

**Pull request merged**
